### ISSUE TITLE
[BUG] After a migration some items reappear to be synced

### ISSUE THREAD

**Issue by mantzas** [djrobstep/migra#78]
After creating a diff and applying it to the new data there are still some items that migra reports as unsynced like:
- create sequence
- alter table add primary key
- alter table add foreign key
- alter table add unique

using docker.io/bitnami/postgresql:10.7.0

**Comment by djrobstep** [comment]
Can you give me an example, so I can reproduce and investigate?

Two schema definitions are all that are required.

**Comment by mantzas** [comment]
this was done with a empty db as a target, does that help?

**Comment by djrobstep** [comment]
yeah that's fine, just give me a dump of the other schema

**Comment by mantzas** [comment]
do you have another way to send the schema?

**Comment by djrobstep** [comment]
if you're concerned about revealing your actual schema, just cut it down to the most minimal example you can, and rename the entity names as necessary

**Comment by mantzas** [comment]
[db.zip](https://github.com/djrobstep/migra/files/2970774/db.zip)

checked and the behavior is the same

**Comment by djrobstep** [comment]
Hmm, I've tried your example, and it syncs fine. Once I apply the migration, there are no more changes present. 

Are you sure you have applied the generated migration? To be clear, the command line tool doesn't change the database itself. It just generates an output script.

**Comment by mantzas** [comment]
yes, I created the change file, opened dbeaver and ran it. 
when I compared again they had a diff. took the diff again to dbeaver ran it but postgres complained that I try to create stuff that already exists.

**Comment by mantzas** [comment]
I have noticed that the migration SQL does not have any user ownership like the original DB. might that play a role?

**Comment by mantzas** [comment]
I have run the procedure from the beginning suing another docker container postgres:10.6 instead of the docker.io/bitnami/postgresql:10.7.0.
With this setup I got it to work.

**Comment by mantzas** [comment]
The privileges flag `--with-privileges` seems not to generate the needed statements for the target DB. 
The source table is created with:
`ALTER TABLE public.specification OWNER TO rems;
GRANT INSERT, SELECT, UPDATE, DELETE ON TABLE public.specification TO rems;
`
but the migration script does not include the above changes with breaks the application that runs with the above role...

**Comment by gregplaysguitar** [comment]
FWIW I just had a similar issue - migra kept telling me to recreate a trigger function which was already there, but not other similar functions I'd defined earlier. After tearing my hair out for a few hours, I realised it only happened when running migra in docker (my own image, but identical to the official one). I forced a rebuild of the docker container and now it works... so, no idea what was going on but I don't think it was migra's fault - hopefully that helps someone else who finds this issue!
